include/wx/msw/ole/activex.h

```diff
@@ -68,76 +69,48 @@ class FrameSite;
 //
 //---------------------------------------------------------------------------
 
-template<typename I>
-class wxAutoOleInterface
+template <class T>
+class wxAutoOleInterface : public wxCOMPtr<T>
 {
+    typedef wxCOMPtr<T> base;
 public:
-    typedef I Interface;
-
-    explicit wxAutoOleInterface(I *pInterface = NULL) : m_interface(pInterface)
-        {}
-    wxAutoOleInterface(REFIID riid, IUnknown *pUnk) : m_interface(NULL)
-        { QueryInterface(riid, pUnk); }
-    wxAutoOleInterface(REFIID riid, IDispatch *pDispatch) : m_interface(NULL)
-        { QueryInterface(riid, pDispatch); }
-    wxAutoOleInterface(REFCLSID clsid, REFIID riid) : m_interface(NULL)
-        { CreateInstance(clsid, riid); }
-    wxAutoOleInterface(const wxAutoOleInterface& ti) : m_interface(NULL)
-        { operator=(ti); }
-
-    wxAutoOleInterface& operator=(const wxAutoOleInterface& ti)
+    wxAutoOleInterface() {}
+    wxAutoOleInterface(REFIID riid, IUnknown* pUnk)
     {
-        if ( ti.m_interface )
-            ti.m_interface->AddRef();
-        Free();
-        m_interface = ti.m_interface;
-        return *this;
+        QueryInterface(riid, pUnk);
     }
 
-    wxAutoOleInterface& operator=(I*& ti)
+    wxAutoOleInterface& operator=(T*& ti)
     {
         Free();
-        m_interface = ti;
+        base::reset(ti);
```

Comment:
Why do we call `Free()` which uses `reset()` already? I.e. shouldn't we just call `reset(ti)` directly?